smart_contract/executor/src/default/permissions.rs

```diff
@@ -43,23 +48,23 @@ macro_rules! declare_tokens {
         }
 
         impl TryFrom<&$crate::data_model::permission::Permission> for AnyPermission {
-            type Error = $crate::permission::PermissionConversionError;
+            type Error = $crate::ConvertDataModelObjectError;
 
             fn try_from(token: &$crate::data_model::permission::Permission) -> Result<Self, Self::Error> {
-                match token.definition_id().as_ref() { $(
+                match token.id().borrow() { $(
```

Comment:
```suggestion
                match token.id().as_ref() { $(
```

you shouldn't use `borrow` manually. It is just for use in `HashMap`s

Reply:
Replaced with `token.id().name().as_ref()`